packages/contracts-bedrock/src/safe/DeputyPauseModule.sol

```diff
@@ -0,0 +1,221 @@
+    /// @notice Error message for nonce reuse.
+    error DeputyPauseModule_NonceAlreadyUsed();
+
+    /// @notice Error message for failed transaction execution.
+    error DeputyPauseModule_ExecutionFailed(string);
+
+    /// @notice Emitted when the SuperchainConfig is paused.
+    /// @param identifier Identifier of the user/action that triggered the pause.
+    event Paused(string identifier);
+
+    /// @notice Struct for the Pause action.
+    /// @custom:field expiry Signature expiry timestamp.
+    /// @custom:field nonce Signature nonce.
+    struct PauseMessage {
+        uint256 expiry;
+        bytes32 nonce;
+    }
+
+    /// @notice Struct for the DeputyAuth action.
+    /// @custom:field deputy Address of the deputy account.
+    struct DeputyAuthMessage {
+        address deputy;
+    }
+
+    /// @notice Foundation Safe.
+    Safe internal immutable FOUNDATION_SAFE;
+
+    /// @notice DeputyGuardianModule used by the Security Council Safe.
+    IDeputyGuardianModule internal immutable DEPUTY_GUARDIAN_MODULE;
+
+    /// @notice Address of the deputy account.
+    address internal immutable DEPUTY;
+
+    /// @notice Maximum signature validity in seconds.
+    uint256 internal immutable MAX_SIGNATURE_VALIDITY_SECONDS;
+
+    /// @notice Typehash for the Pause action.
+    bytes32 internal constant PAUSE_MESSAGE_TYPEHASH = keccak256("PauseMessage(uint256 expiry,bytes32 nonce)");
+
+    /// @notice Typehash for the DeputyAuth message.
+    bytes32 internal constant DEPUTY_AUTH_MESSAGE_TYPEHASH = keccak256("DeputyAuthMessage(address deputy)");
+
+    /// @notice Used nonces.
+    mapping(bytes32 => bool) public usedNonces;
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0-beta.1
+    string public constant version = "1.0.0-beta.1";
+
+    /// @param _foundationSafe Address of the Foundation Safe.
+    /// @param _deputyGuardianModule Address of the DeputyGuardianModule used by the SC Safe.
+    /// @param _maxSignatureValiditySeconds Maximum signature validity in seconds.
+    /// @param _deputy Address of the deputy account.
+    /// @param _deputySignature Signature from the deputy verifying that the account is an EOA.
+    constructor(
+        Safe _foundationSafe,
+        IDeputyGuardianModule _deputyGuardianModule,
+        uint256 _maxSignatureValiditySeconds,
+        address _deputy,
+        bytes memory _deputySignature
+    )
+        EIP712()
+    {
+        // Check that the deputy is an EOA.
+        // We do not support EIP-1271 signatures here.
+        bytes32 digest = _hashTypedData(keccak256(abi.encode(DEPUTY_AUTH_MESSAGE_TYPEHASH, DeputyAuthMessage(_deputy))));
+        if (ECDSA.recover(digest, _deputySignature) != _deputy) {
+            revert DeputyPauseModule_InvalidDeputy();
+        }
+
+        // Zero second validity would mean that signatures are never valid.
+        if (_maxSignatureValiditySeconds == 0) {
+            revert DeputyPauseModule_SignatureValidityTooShort();
+        }
```

Comment:
Should we put a nonzero min bound here similar to the upper bound? In practice, what duration do you have in mind, e.g. would 1 day be an ok min?

Reply:
Hmmm good question. I think 1 day is probably a good min. You can always just sign shorter expiry timestamps if you really want to.